### ISSUE TITLE
metamorphic: TestMeta failed

### ISSUE THREAD

**Issue by jbowens** [cockroachdb/pebble#1950]
I was running the metamorphic tests on my gceworker on SHA dc98ad3ab7f7ddbb1a832a53685093fc7a884195, which surfaced the following failure:

```
===== SEED =====
1663010036498345365
===== DIFF =====
_meta/220912-191356.498/{standard-000,standard-005}
@@ -672,11 +672,11 @@
 db.Merge("mbboldrync@5", "onvzgzhuvpz") // <nil> #671
 iter2.SeekLT("hgzwxlnnxt@3", "") // [false] <nil> #672
 iter1.SeekPrefixGE("iyzpptyg@1") // [false] <nil> #673
 iter5.SetBounds("", "") // <nil> #674
 db.Get("hgzwxlnnxt@4") // ["moimciwvgolhdwjyvjv"] <nil> #675
-iter5.SeekLT("tfkghlkguka", "") // [true,"srpvhivggb",<no point>,["srpvhivggb","uttjuiwtmor")=>{"@4"="twxhzzjvsjftzd"}] <nil> #676
+iter5.SeekLT("tfkghlkguka", "") // [true,"srpvhivggb",<no point>,["srpvhivggb","uttjuiwtmor")=>{"@4"="twxhzzjvsjftzd"}*] <nil> #676
 iter2.Next("") // [true,"hgzwxlnnxt@3","snfwphjdsyjgrcwykba",<no range>] <nil> #677
 db.Set("lhdwk@4", "skz") // <nil> #678
 iter2.SeekLT("kdkegql@4", "") // [true,"hgzwxlnnxt@3","snfwphjdsyjgrcwykba",<no range>] <nil> #679
 iter1.SetBounds("fved@3", "fved@3") // <nil> #680
 iter5.SeekGE("rdzo@2", "") // [true,"srpvhivggb",<no point>,["srpvhivggb","uttjuiwtmor")=>{"@4"="twxhzzjvsjftzd"}] <nil> #681
 ```

**Comment by bananabrick** [comment]
Doesn't fail for me on my gceworker on the latest commit from master.

`go test -run TestMeta -seed 1663010036498345365`

Latest commit:
```
commit 10b967f62b36bda8a3efce493dee08320e031eaa (HEAD -> master, upstream/master)
Author: Jackson Owens <jackson@cockroachlabs.com>
Date:   Mon Sep 12 10:31:30 2022 -0400

    db: ensure RangeKeyChanged is true after no-op SetBounds
    
    Previously, if a call to SetBounds hit the no-op optimization because the
    bounds were identical, a subsequent seek that discovered the same range key as
    the last valid iterator position would incorrectly report RangeKeyChanged() as
    false.
    
    Fix #1947.
    ```

**Comment by jbowens** [comment]
fails for me locally, although nondeterministically:

```
go test -v -run 'TestMeta$' --seed 1663010036498345365 -tags invariants --keep
```
